nf_core/modules/lint/environment_yml.py

```diff
@@ -82,41 +103,81 @@ def environment_yml(module_lint_object: ComponentLint, module: NFCoreComponent,
             )
 
         if valid_env_yml:
-            # Check that the dependencies section is sorted alphabetically
-            def sort_recursively(obj):
-                """Simple recursive sort for nested structures."""
-                if isinstance(obj, list):
-
-                    def get_key(x):
-                        if isinstance(x, dict):
-                            # For dicts like {"pip": [...]}, use the key "pip"
-                            return (list(x.keys())[0], 1)
-                        else:
-                            # For strings like "pip=23.3.1", use "pip" and for bioconda::samtools=1.15.1, use "bioconda::samtools"
-                            return (str(x).split("=")[0], 0)
-
-                    return sorted([sort_recursively(item) for item in obj], key=get_key)
-                elif isinstance(obj, dict):
-                    return {k: sort_recursively(v) for k, v in obj.items()}
-                else:
-                    return obj
-
-            sorted_dependencies = sort_recursively(env_yml["dependencies"])
-
-            # Direct comparison of sorted vs original dependencies
-            if sorted_dependencies == env_yml["dependencies"]:
-                module.passed.append(
-                    (
+            # Sort dependencies if they exist
+            if "dependencies" in env_yml:
+                dicts = []
+                others = []
+
+                for term in env_yml["dependencies"]:
+                    if isinstance(term, dict):
+                        dicts.append(term)
+                    else:
+                        others.append(term)
+
+                # Sort non-dict dependencies with special handling for pip
+                def sort_key(x):
+                    # Convert to string for comparison
+                    str_x = str(x)
+                    # If it's a pip package (but not pip itself), put it after other conda packages
+                    if str_x.startswith("pip=") or str_x == "pip":
```

Comment:
is it only `pip` that will have to be sorted inside itself? or are there other cases that we should take into account?